cgi/cgiutils.c

```diff
@@ -1116,7 +1116,7 @@ static inline char* encode_character(char in, char *outcp, int outcp_len, char *
 	if (out_len + 6 >= outcp_len - 1)
```

Comment:
I don't remember exactly what the reasoning was for choosing "+6", but I suspect we're guarding against segfault here. Can you test that this is still sufficient by adding some hosts and services with very long names using the CJK character set?

Reply:
```
		else if ((unsigned char)*incp > 0x7f)
			/* pass through UTF-8 characters */
			*outcp++ = *incp;

		/* for simplicity, all other chars represented by their numeric value */
		else
			outcp = encode_character(*incp, outcp, output_max, *output);
```
Anything past 7f (127) is not encoded so I don't think we have to worry about utf characters like ϴ

Reply:
okay, nice! I'm hearing that it should now be
```suggestion
	if (out_len + 7 >= outcp_len - 1)
```
since the array needs to fit `&#127;\0`

Reply:
Well it would only need 6 because you'd want `~~` to be `&#126;&#126;\0` and not `&#126;\0&#126;\0`

If anything it should be changed to 5 (although that doesn't initially make a whole lot of sense) or the conditional changed to `if (out_len + 6 > outcp_len - 1)` (which makes more sense)

I tested this with a plugin that outputs `~~~~` and added the ~ character to the list of characters that should be encoded. It will always strip off the last ~. 

`~~~~` would create an output buffer of size 25. For the final encoding:
```
outcp = output[18]
outcp_len = 25
out_len = output[18] - output[0] = 18
18 + 6 = 24
25 - 1 = 24
```
And it fails, even though there is enough space left to put six characters

Reply:
You can also test with this plugin
```
echo '^'
exit 0
```